packages/ptokens-assets-evm/src/ptokens-evm-provider.ts

```diff
@@ -205,24 +197,37 @@ export class pTokensEvmProvider implements pTokensAssetProvider {
         return false
       }
     }, _pollingTime)
+    if (!receipt) throw new Error('Polling function stopped unexpectedly. No receipt found')
     return receipt.transactionHash.toString()
   }
 
+  /**
+   * Wait for tx receipt.
+   * @param _txHash - Transcation hash of the searched receipt.
+   * @returns TransactionReceipt of _txHash.
+   */
+  async waitForTransactionReceipt(_txHash: `0x${string}`): Promise<TransactionReceipt> {
+    const receipt: TransactionReceipt = await this._publicClient.waitForTransactionReceipt({ hash: _txHash})
+    if (!receipt) throw new Error('No receipt found')
+    return receipt
+  }
+
   protected async _pollForHubOperation(
     _hubAddress: string,
     _eventName: EVENT_NAMES,
     _fromBlock: bigint,
     _operationId: string,
     _pollingTime = 1000,
-  ) {
-    let log
+  ): Promise<Log> {
+    let log: Log | undefined
     // eslint-disable-next-line @typescript-eslint/no-unsafe-call
     await polling(async () => {
       try {
+        log = undefined
         const logs = await this._publicClient.getLogs({
```

Comment:
Do we really need this? I remember you said time ago that viem has a function to poll automatically?

It should be https://viem.sh/docs/contract/watchContractEvent.html#watchcontractevent.
Interesting:
```
watchContractEvent will attempt to create an [Event Filter](https://viem.sh/docs/contract/createContractEventFilter.html) and listen to changes to the Filter per polling interval, however, if the RPC Provider does not support Filters (ie. eth_newFilter), then watchContractEvent will fall back to using [getLogs](https://viem.sh/docs/actions/public/getLogs.html) instead.
```